universe/Universe.cpp

```diff
@@ -2196,10 +2196,10 @@ void Universe::ExecuteEffects(const Effect::TargetsCauses& targets_causes,
     // grouping targets causes by effects group
     // sorting by effects group has already been done in GetEffectsAndTargets()
     // FIXME: GetEffectsAndTargets already produces this separation, exploit that
-    std::vector<std::pair<Effect::EffectsGroup*, Effect::TargetsCauses> > dispatched_targets_causes;
+    std::map<int, std::vector<std::pair<Effect::EffectsGroup*, Effect::TargetsCauses> > > dispatched_targets_causes;
```

Comment:
Consider making the map key value a float, to make it easier to put new effectsgroups at priorities between existing priorities?

Also, my expectation would be that higher priorities would go first, but as noted above, default map sorting and iteration would go from low to high values. Could add a sort function here so the map is iterated through from high to low values of the key.